pages/api/templates.ts

```diff
@@ -13,6 +14,13 @@ const templates = async (req: NextApiRequest, res: NextApiResponse) => {
     const session = await isAdmin({ req });
     const response = await crudTemplates({ ...req.body, method: req.method, session });
     if (response) {
+      if (session && session.user.id && (req.method === "POST" || "PUT" || "DELETE")) {
```

Comment:
I haven't executed this line of code, but I think you're actually missing the comparison against `req.method`.

```(req.method === "POST" || req.method === "PUT" || req.method === "DELETE")```